conda_build/metadata.py

```diff
@@ -279,15 +279,18 @@ def check_fields(meta, strictness=.9):
     try:
         _ = _meta['package']['name']
         _ = _meta['package']['version']
-    except IndexError:
-        raise ValueError("Invalid recipe")
+    except KeyError:
+        sys.exit("Invalid meta.yaml: Missing package/name or package/version.")
```

Comment:
My gut reaction was also to prefer exceptions instead of `sys.exit`, but the rest of the code base really isn't set up for that.  Throughout the rest of this file (`metadata.py`), all errors use `sys.exit` instead of `raise`.  If you use `raise` here, then the user sees a crash report, telling them to post the traceback to github.

If you want to change these from `sys.exit` to `raise`, you'll need to define a special type of exception that represents errors that aren't bugs, and handling logic at the top-level somewhere (e.g. `main_build.py`).  But those changes seemed outside the scope of what I was aiming for in this PR, so I decided to just stay consistent with the conventions already used in this file (and elsewhere).